src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/NotificationHandlerConfiguration.java

```diff
@@ -3745,6 +3774,46 @@ public PreSubmitCallbackHandler<AsylumCase> submitAppealLrHoWaysToPayPaPayNowNot
         );
     }
 
+    @Bean
+    public PreSubmitCallbackHandler<AsylumCase> aipDisposeUnderRule31Or32AppelantNotificationHandler(
+        @Qualifier("aipDisposeUnderRule31Or32AppelantNotificationGenerator")
+        List<NotificationGenerator> notificationGenerators) {
+
+        return new NotificationHandler(
+            (callbackStage, callback) -> {
+                AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+                return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+                    && callback.getEvent() == Event.DECIDE_FTPA_APPLICATION
+                    && isAipJourney(asylumCase)
+                    && isFtpaDecisionOutcomeTypeUnderRule31OrRule32(asylumCase);
+
+            },
+            notificationGenerators,
+            getErrorHandler()
+        );
+    }
+
+    private boolean isFtpaDecisionOutcomeTypeUnderRule31OrRule32(AsylumCase asylumCase) {
+        List<String> ftpaDecisionOutcomeTypes = List.of(
+            FtpaDecisionOutcomeType.FTPA_REMADE31.toString(),
+            FtpaDecisionOutcomeType.FTPA_REMADE32.toString()
+        );
+
+        boolean appellantDecisionOutcomeIsRule31Or32 = asylumCase
```

Comment:
I think we need to check the latest data from 'ftpaApplicantType' field to see which outcome type field to read from. 'RESPONDENT' or 'APPELLANT'